### ISSUE TITLE
[2주차] Optimization, Regression 과제 제출 (~ 1/30)

### ISSUE THREAD

**Issue by tobigs-datamarket** [tobigs-datamarket/tobigs-21st#19]
# Week2. Optimization, Regression

## Part 1. Optimization
아래 항목들을 반드시 포함하여 과제를 제출해주시기 바랍니다!

- [ ] `Week2/Optimization/wk2_optimization_assignment.ipynb`의 마크다운, 코드 빈칸 채우기
- [ ] 완성된 함수로 주어진 데이터에 대해 gradient descent(경사하강법) 진행하기
- [ ] Hyperparameter를 바꿔가며 실행하기
- [ ] 완성한 코드에 대해 상세하게 주석 달기

## Part 2. Regression
Regression 과제는 3개의 항목으로 구성되어 있습니다!

[과제 1]

`Week2/Regression/[투빅스 21기 정규세션] 2주차 Regression_과제 1.ipynb`에
- [ ] LSE(Least Square Method) normal equation 구현하기
- [ ] MSE 구현하기

[과제 2] 회귀분석 - Used Car Price Prediction

`Week2/Regression/[투빅스 21기 정규세션] 2주차 Regression_과제 2.ipynb`에
- [ ] Ch 1, Ch 2를 토대로 자유롭게 회귀분석과 회귀진단하기
- [ ] 주석으로 설명 및 근거 자세하게 기록하기

[과제 3] 로지스틱 회귀분석 - Credit Card Fraud Detection

`Week2/Regression/[투빅스 21기 정규세션] 2주차 Regression_과제 3.ipynb`에
- [ ] `sklearn` 패키지를 활용해 로지스틱 회귀분석하기
- [ ] 성능지표 계산 및 해석하기
  - [ ] `sklearn`의 mean accuracy, F1 score 등 체크하기
  - [ ] Confusion matrix의 TP, FP, FN, TN 체크하기
- [ ] 성능 개선 시도하기
  - [ ]  어떤 성능 지표를 기준으로 했는지, 해당 지표를 왜 선택했는지 등 설명하기
- [ ] 주석으로 설명 및 근거 자세하게 기록하기

## 제출 방법
1. 본인의 이름으로 branch를 생성합니다.
2. `main`을 본인의 branch로 pull합니다.
6. **본인의 branch에서** 과제 디렉토리에 본인의 이름으로 디렉토리를 생성합니다.
7. 과제를 수행합니다. 디렉토리 구조:
   ```
   tobigs-21st
   └── Week1
       ├── EDA
       │   └── 김투빅
       │       └── example.md # 과제 파일
       └── Framework
           └── 김투빅
               └── example.md # 과제 파일
   ```
8. 변경된 본인의 branch를 commit, push합니다.
9. `이름(기수) - n주차 과제`라는 타이틀로 `Pull Request`를 등록합니다. 예) 김투빅(21기) - 1주차 과제

## 제출 기한
⏰ ~ 1/30(화) 23:59
